modules/broker-auth/src/main/java/io/ballerina/messaging/broker/auth/authorization/handler/AuthorizationHandler.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+package io.ballerina.messaging.broker.auth.authorization.handler;
+
+import io.ballerina.messaging.broker.auth.AuthManager;
+import io.ballerina.messaging.broker.auth.UsernamePrincipal;
+import io.ballerina.messaging.broker.auth.authorization.Authorizer;
+import io.ballerina.messaging.broker.auth.authorization.authorizer.rdbms.resource.AuthResource;
+import io.ballerina.messaging.broker.auth.authorization.enums.ResourceActions;
+import io.ballerina.messaging.broker.auth.authorization.enums.ResourceAuthScopes;
+import io.ballerina.messaging.broker.auth.authorization.enums.ResourceTypes;
+import io.ballerina.messaging.broker.auth.exception.BrokerAuthDuplicateException;
+import io.ballerina.messaging.broker.auth.exception.BrokerAuthException;
+import io.ballerina.messaging.broker.auth.exception.BrokerAuthNotFoundException;
+import io.ballerina.messaging.broker.auth.exception.BrokerAuthServerException;
+import io.ballerina.messaging.broker.common.ResourceNotFoundException;
+
+import javax.security.auth.Subject;
+
+/**
+ * Class handles authorization for broker actions.
+ */
+public class AuthorizationHandler {
+
+    private Authorizer authorizer;
+    private UsernamePrincipal userPrincipal;
+
+    public AuthorizationHandler(AuthManager authManager, Subject subject) {
+        authorizer = authManager.getAuthorizer();
+        this.userPrincipal = (UsernamePrincipal) subject.getPrincipals().iterator().next();
+    }
+
+    /**
+     * Handle given auth scope and auth resource authorization.
+     *
+     * @param brokerAuthScope auth scope
+     * @param resourceType    resource type
+     * @param resourceName    resource name
+     * @param action          action
+     * @throws BrokerAuthException throws if error occurs while authorizing resource.
+     */
+    public void handle(ResourceAuthScopes brokerAuthScope, ResourceTypes resourceType, String resourceName,
```

Comment:
What if we pass the subject as an argument here? In that case we can just pass a reference to a authorizationHandler instance in SecuredBrokerImpl rather than creating a new AuthorizationHandler each time we create a SecuredBrokerImpl.

Reply:
Fixed in 810c187